support/scripts/multiboot2.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+# Licensed under the BSD-3-Clause License (the "License").
+# You may not use this file except in compliance with the License.
+
+import os
+import argparse
+from struct import unpack
+
+# Default global values
+ELF_MAGIC = {
+    "i386": b"\x7fELF\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00",
+    "x86_64": b"\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00",
+}
+
+ELF_MACHINE = {
+    "i386": 3,  # EM_386
+    "x86_64": 62,  # EM_X86_64
+}
+
+ELF64_EHDR_LEN = 64
+
+ELF64_PHDR_LEN = 56
+
+ELF64_SHDR_LEN = 64
+
+# The fixed-size part of the Multiboot header is of size 16
+# (Magic(4) + Architecture(4) + Header length(4) + Checksum(4)).
+# The rest of the header is variable in size depending on the chosen tags
+MULTIBOOT_HEADER_BASE_SIZE = 16
+
+MULTIBOOT_HEADER_MAGIC = 0xE85250D6
+MULTIBOOT_ARCHITECTURE_I386 = 0
+MULTIBOOT_LOAD_PREFERENCE_LOW = 1
+PAGE_SIZE = 4096
+
+MULTIBOOT_HEADER_TAG_OPTIONAL = 1
+
+MULTIBOOT_HEADER_TAG_END = 0
```

Comment:
Let's only declare the tags that are used. And for those that are used give a brief description/comment for what they are used for and if there is any special treatment from GRUB2 worth mentioning (that is not mandated by the spec).